python-checks/src/test/resources/checks/tests/assertOnDissimilarTypes.py

```diff
@@ -22,16 +22,20 @@ def __ne__(self, other):
 
 class MyTest(unittest.TestCase):
   def test_assert_is_non_compliant(self):
-    self.assertEqual(first="string", second=True) # Noncompliant
     myString = "toto"
     myInt = 3
     a = A()
-#   ^^^^^^^> {{Last assignment of "a"}}
     b = B()
-#   ^^^^^^^> {{Last assignment of "b"}}
-
-    self.assertIs(a, b) # Noncompliant {{Change this assertion to not compare dissimilar types (A and B).}}
-#                 ^^^^
+    aInd = A()
+#   ^^^^^^^^^^> {{Last assignment of "aInd"}}
+    bInd = B()
+#   ^^^^^^^^^^> {{Last assignment of "bInd"}}
+
+    self.assertIs(aInd, bInd) # Noncompliant {{Change this assertion to not compare dissimilar types (A and B).}}
+#                 ^^^^^^^^^^
+    aInd = A()
+    self.assertEqual(first="string", second=True) # Noncompliant
+    self.assertIs(msg="my message", first="myString", second=a) # Noncompliant
```

Comment:
For the sake of completeness, I'd probably add a compliant test where the arguments are in an unexpected order where we don't raise an FP because we correctly understand the keyword arguments.